api/v1alpha3/azuremachine_validation.go

```diff
@@ -165,6 +174,21 @@ func ValidateManagedDisk(old, new ManagedDisk, fieldPath *field.Path) field.Erro
 	return allErrs
 }
 
+// ValidateManagedDisks validates updates to Data disks' managed disk field.
+func ValidateManagedDisks(old, new []DataDisk, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	for i, disk := range new {
```

Comment:
here we should also check for len() of new and old so we don't run into an index out of range panic if they don't have equal lengths

Reply:
If the len() mismatch, what is the error message that we want to return?